api/v1beta1/manila_webhook.go

```diff
@@ -332,3 +338,45 @@ func (spec *ManilaSpec) ValidateManilaTopology(basePath *field.Path, namespace s
 	}
 	return allErrs
 }
+
+// ValidateBaseParams -
+func (spec *ManilaSpec) ValidateBaseParams(basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if spec.RabbitMqClusterName == "" {
+		rabbitPath := basePath.Child("rabbitMqClusterName")
+		allErrs = append(allErrs,
+			field.Invalid(rabbitPath, "rabbitMqClusterName", "RabbitMqClusterName is empty"))
+	}
+	if spec.MemcachedInstance == "" {
+		mcPath := basePath.Child("memcachedInstance")
+		allErrs = append(allErrs,
+			field.Invalid(mcPath, "memcachedInstance", "MemcachedInstance is empty"))
+	}
+	if spec.DatabaseAccount == "" {
+		dbAcc := basePath.Child("databaseAccount")
+		allErrs = append(allErrs,
+			field.Invalid(dbAcc, "databaseAccount", "DatabaseAccount is empty"))
+	}
+	return allErrs
+}
+
+// ValidateBaseParams -
+func (spec *ManilaSpecCore) ValidateBaseParams(basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if spec.RabbitMqClusterName == "" {
+		rabbitPath := basePath.Child("rabbitMqClusterName")
+		allErrs = append(allErrs,
+			field.Invalid(rabbitPath, "rabbitMqClusterName", "RabbitMqClusterName is empty"))
+	}
+	if spec.MemcachedInstance == "" {
+		mcPath := basePath.Child("memcachedInstance")
+		allErrs = append(allErrs,
+			field.Invalid(mcPath, "memcachedInstance", "MemcachedInstance is empty"))
+	}
+	if spec.DatabaseAccount == "" {
+		dbAcc := basePath.Child("databaseAccount")
+		allErrs = append(allErrs,
+			field.Invalid(dbAcc, "databaseAccount", "DatabaseAccount is empty"))
+	}
+	return allErrs
```

Comment:
could refactor this into a method that we can call from both these overloaded forms?

Reply:
I can't atm! Mostly because we need some refactor of the SpecBase to have a single one.
We can plan for this work, but as a dedicated follow up.
(Note that for the same reason I couldn't do it for topologies as well [1])

[1] https://github.com/openstack-k8s-operators/manila-operator/blob/main/api/v1beta1/manila_webhook.go#L274